test/util/util.go

```diff
@@ -128,11 +134,69 @@ func DeleteAllJobsInNamespace(ctx context.Context, c client.Client, ns *corev1.N
 	return nil
 }
 
+func DeleteAllPodsInNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace) error {
+	err := c.DeleteAllOf(ctx, &corev1.Pod{}, client.InNamespace(ns.Name))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	lst := corev1.PodList{}
+	err = c.List(ctx, &lst, client.InNamespace(ns.Name))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if len(lst.Items) == 0 {
+		return nil
+	}
+
+	for _, p := range lst.Items {
+		copy := p.DeepCopy()
+		if controllerutil.RemoveFinalizer(copy, pod.PodFinalizer) {
+			err = c.Update(ctx, copy)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
 func DeleteWorkloadsInNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace) error {
 	if err := c.DeleteAllOf(ctx, &kueue.Workload{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	return nil
+
+	err := wait.ExponentialBackoffWithContext(ctx,
```

Comment:
Since we are already using `gomega` in this package, prefer to use `Eventually` here.

Why don't we need the same retry logic for deleting Pods?